src/gui/systray.cpp

```diff
@@ -285,6 +285,103 @@ void Systray::destroyEditFileLocallyLoadingDialog()
     _editFileLocallyLoadingDialog = nullptr;
 }
 
+bool Systray::raiseDialogs()
+{
+    return raiseFileDetailDialogs();
+}
+
+bool Systray::raiseFileDetailDialogs(const QString &localPath)
+{
+    if(_fileDetailDialogs.empty()) {
+        return false;
+    }
+
+    auto it = _fileDetailDialogs.begin();
+    while (it != _fileDetailDialogs.end()) {
+        const auto dialog = *it;
+        auto liveDialog = dialog != nullptr;
+
+        if (liveDialog && !dialog->isVisible()) {
+            destroyDialog(dialog);
+            liveDialog = false;
+        }
+
+        if (liveDialog && (localPath.isEmpty() || dialog->property("localPath").toString() == localPath)) {
+            dialog->show();
+            dialog->raise();
+            dialog->requestActivate();
+
+            ++it;
+            continue;
+        }
+
+        it = _fileDetailDialogs.erase(it);
+        continue;
+    }
+
+    // If it is empty then we have raised no dialogs, so return false (and viceversa)
+    return !_fileDetailDialogs.empty();
+}
+
+void Systray::createFileDetailsDialog(const QString &localPath)
+{
+    if (raiseFileDetailDialogs(localPath)) {
+        qCDebug(lcSystray) << "Reopening an existing file details dialog for " << localPath;
+        return;
+    }
+
+    qCDebug(lcSystray) << "Opening new file details dialog for " << localPath;
+
+    if (!_trayEngine) {
+        qCWarning(lcSystray) << "Could not open file details dialog for" << localPath << "as no tray engine was available";
+        return;
+    }
+
+    const auto folder = FolderMan::instance()->folderForPath(localPath);
+    if (!folder) {
+        qCWarning(lcSystray) << "Could not open file details dialog for" << localPath << "no responsible folder found";
+        return;
+    }
+
+    const QVariantMap initialProperties{
+        {"accountState", QVariant::fromValue(folder->accountState())},
+        {"localPath", localPath},
+    };
+
+    QQmlComponent fileDetailsDialog(_trayEngine, QStringLiteral("qrc:/qml/src/gui/filedetails/FileDetailsWindow.qml"));
+
+    if (!fileDetailsDialog.isError()) {
+        const auto createdDialog = fileDetailsDialog.createWithInitialProperties(initialProperties);
+        const auto dialog = dynamic_cast<QQuickWindow*>(createdDialog);
```

Comment:
should not you use `qobject_cast`? https://doc.qt.io/qt-6/qobject.html#qobject_cast

Reply:
Now using it